### ISSUE TITLE
Generating cluster region ground truth

### ISSUE THREAD

**Issue by sh7jacobi** [fyangneil/Clustered-Object-Detection-in-Aerial-Image#14]
I want to train model in server, so I can not run:

run "./detectron/ops/add_cluster_annotation.m" to generate cluster ground truth and add it to original object annotation files.
run "./detectron/ops/visdrone2cocoformat.m" to convert VisDrone format annotation to COCO format.

When I run this I have permission denied.

I guess this is code implemented in Matlab. May by there is some python implementation?

**Comment by fyangneil** [comment]
you can run it locally and transfer the ground truth file to server.
________________________________

**Comment by sh7jacobi** [comment]
Ok, thank you, I thought about the same solution. Could you send me some material about how to generate clustered ground truth?27.01.2021, 17:49, "fyangneil" <notifications@github.com>:
you can run it locally and transfer the ground truth file to server.
________________________________

**Comment by fyangneil** [comment]
run meanshift clustering algorithm on location (x,y) of ground truth bounding box.
________________________________

**Comment by fyangneil** [comment]
then use boxes belong to the same cluster to yield a larger box as cluster box
________________________________

**Comment by sh7jacobi** [comment]
I have one more question. In detectron it is need to convert data to cocoformat. I see that in visdrone2cocoformat.m it is with val data, but in add_cluster_annotation.m data is saved in old format.27.01.2021, 17:59, "fyangneil" <notifications@github.com>:
then use boxes belong to the same cluster to yield a larger box as cluster box
________________________________

**Comment by sh7jacobi** [comment]
And why is clustered bboxes written in one file with simple bboxes? I would like to know how I should set data with clustered bboxes that I could train.28.01.2021, 10:37, "Якуб Шарафутдинов" <dazgenovich2012@yandex.ru>:I have one more question. In detectron it is need to convert data to cocoformat. I see that in visdrone2cocoformat.m it is with val data, but in add_cluster_annotation.m data is saved in old format.27.01.2021, 17:59, "fyangneil" <notifications@github.com>:
then use boxes belong to the same cluster to yield a larger box as cluster box
________________________________

**Comment by sh7jacobi** [comment]
Do you have pretrained weights?28.01.2021, 11:45, "Якуб Шарафутдинов" <dazgenovich2012@yandex.ru>:And why is clustered bboxes written in one file with simple bboxes? I would like to know how I should set data with clustered bboxes that I could train.28.01.2021, 10:37, "Якуб Шарафутдинов" <dazgenovich2012@yandex.ru>:I have one more question. In detectron it is need to convert data to cocoformat. I see that in visdrone2cocoformat.m it is with val data, but in add_cluster_annotation.m data is saved in old format.27.01.2021, 17:59, "fyangneil" <notifications@github.com>:
then use boxes belong to the same cluster to yield a larger box as cluster box
________________________________

**Comment by fyangneil** [comment]
the cluster box is not in a simple file, it is stored as a class of object with other normal object boxes. Sorry, since it is long time ago, I do not have any weight file in hand.
________________________________

**Comment by sh7jacobi** [comment]
I have fail when I run train, I think that is problem with version of cudnn. What version of cudnn you used?28.01.2021, 18:28, "fyangneil" <notifications@github.com>:
the cluster box is not in a simple file, it is stored as a class of object with other normal object boxes. Sorry, since it is long time ago, I do not have any weight file in hand.
________________________________

**Comment by sh7jacobi** [comment]
Hello, I have an error on train on first epoch: ...INFO loader.py: 235:   [50/64]INFO loader.py: 235:   [51/64]INFO loader.py: 235:   [52/64]INFO loader.py: 235:   [53/64]INFO loader.py: 235:   [53/64]INFO loader.py: 235:   [53/64]INFO loader.py: 235:   [53/64]INFO loader.py: 235:   [55/64]INFO loader.py: 235:   [56/64]INFO loader.py: 235:   [57/64]INFO loader.py: 235:   [57/64]INFO loader.py: 235:   [58/64]INFO loader.py: 235:   [59/64]INFO loader.py: 235:   [61/64]INFO loader.py: 235:   [61/64]INFO loader.py: 235:   [61/64]INFO loader.py: 235:   [61/64]START IterTic !INFO detector.py: 753: Changing learning rate 0.000000 -> 0.000833 at iter 0START RunNet !I0202 13:32:29.233763  2948 context_gpu.cu:321] GPU 0: 811 MBI0202 13:32:29.233817  2948 context_gpu.cu:325] Total: 811 MBterminate called after throwing an instance of 'caffe2::EnforceNotMet'  what():  [enforce fail at conv_op_cudnn.cc:572] status == CUDNN_STATUS_SUCCESS. 8 vs 0. , Error at: /var/lib/jenkins/workspace/caffe2/operators/conv_op_cudnn.cc:572: CUDNN_STATUS_EXECUTION_FAILED Error from operator:input: "gpu_0/data" input: "gpu_0/conv1_w" output: "gpu_0/conv1" name: "" type: "Conv" arg { name: "kernel" i: 7 } arg { name: "exhaustive_search" i: 0 } arg { name: "pad" i: 3 } arg { name: "order" s: "NCHW" } arg { name: "stride" i: 2 } device_option { device_type: 1 cuda_gpu_id: 0 } engine: "CUDNN"*** Aborted at 1612272750 (unix time) try "date -d @1612272750" if you are using GNU date ***PC: @     0x7fdb9b88e428 gsignal*** SIGABRT (@0xb47) received by PID 2887 (TID 0x7fd9b17fa700) from PID 2887; stack trace: ***    @     0x7fdb9bc34390 (unknown)    @     0x7fdb9b88e428 gsignal    @     0x7fdb9b89002a abort    @     0x7fdb9593c84d __gnu_cxx::__verbose_terminate_handler()    @     0x7fdb9593a6b6 (unknown)    @     0x7fdb9593a701 std::terminate()    @     0x7fdb95965d38 (unknown)    @     0x7fdb9bc2a6ba start_thread    @     0x7fdb9b96041d clone    @                0x0 (unknown)Aborted (core dumped)   How I can fix it?

**Comment by sh7jacobi** [comment]
It fails on codeline: worspace.RunNet(model.net.Proto().name)02.02.2021, 16:45, "Якуб Шарафутдинов" <dazgenovich2012@yandex.ru>:Hello, I have an error on train on first epoch: ...INFO loader.py: 235:   [50/64]INFO loader.py: 235:   [51/64]INFO loader.py: 235:   [52/64]INFO loader.py: 235:   [53/64]INFO loader.py: 235:   [53/64]INFO loader.py: 235:   [53/64]INFO loader.py: 235:   [53/64]INFO loader.py: 235:   [55/64]INFO loader.py: 235:   [56/64]INFO loader.py: 235:   [57/64]INFO loader.py: 235:   [57/64]INFO loader.py: 235:   [58/64]INFO loader.py: 235:   [59/64]INFO loader.py: 235:   [61/64]INFO loader.py: 235:   [61/64]INFO loader.py: 235:   [61/64]INFO loader.py: 235:   [61/64]START IterTic !INFO detector.py: 753: Changing learning rate 0.000000 -> 0.000833 at iter 0START RunNet !I0202 13:32:29.233763  2948 context_gpu.cu:321] GPU 0: 811 MBI0202 13:32:29.233817  2948 context_gpu.cu:325] Total: 811 MBterminate called after throwing an instance of 'caffe2::EnforceNotMet'  what():  [enforce fail at conv_op_cudnn.cc:572] status == CUDNN_STATUS_SUCCESS. 8 vs 0. , Error at: /var/lib/jenkins/workspace/caffe2/operators/conv_op_cudnn.cc:572: CUDNN_STATUS_EXECUTION_FAILED Error from operator:input: "gpu_0/data" input: "gpu_0/conv1_w" output: "gpu_0/conv1" name: "" type: "Conv" arg { name: "kernel" i: 7 } arg { name: "exhaustive_search" i: 0 } arg { name: "pad" i: 3 } arg { name: "order" s: "NCHW" } arg { name: "stride" i: 2 } device_option { device_type: 1 cuda_gpu_id: 0 } engine: "CUDNN"*** Aborted at 1612272750 (unix time) try "date -d @1612272750" if you are using GNU date ***PC: @     0x7fdb9b88e428 gsignal*** SIGABRT (@0xb47) received by PID 2887 (TID 0x7fd9b17fa700) from PID 2887; stack trace: ***    @     0x7fdb9bc34390 (unknown)    @     0x7fdb9b88e428 gsignal    @     0x7fdb9b89002a abort    @     0x7fdb9593c84d __gnu_cxx::__verbose_terminate_handler()    @     0x7fdb9593a6b6 (unknown)    @     0x7fdb9593a701 std::terminate()    @     0x7fdb95965d38 (unknown)    @     0x7fdb9bc2a6ba start_thread    @     0x7fdb9b96041d clone    @                0x0 (unknown)Aborted (core dumped)   How I can fix it?
